pkg/controllerutils/upsert.go

```diff
@@ -71,15 +70,12 @@ func transition(existing, desired runtime.Object, transitionFuncs []TransitionFu
 //
 // If the desired object status is semantically equal
 // to the existing object status, the update is skipped.
-func UpdateStatus(ctx context.Context, c client.Client, obj runtime.Object) (controllerutil.OperationResult, error) {
-	key, err := client.ObjectKeyFromObject(obj)
-	if err != nil {
-		return controllerutil.OperationResultNone, err
-	}
+func UpdateStatus(ctx context.Context, c client.Client, obj client.Object) (controllerutil.OperationResult, error) {
+	key := client.ObjectKeyFromObject(obj)
 
 	existing := obj.DeepCopyObject()
 
-	if err := c.Get(ctx, key, existing); err != nil {
+	if err := c.Get(ctx, key, obj); err != nil {
```

Comment:
why are we switching `obj` and `existing` here?

Reply:
if this is a bug, we should have a test that would have caught this swap

Reply:
Agreed, this requires the same fix as the `Upsert` function above

Reply:
Yeah that's strange, I have this locally: 

`
	existing := obj.DeepCopyObject().(client.Object)

	if err := c.Get(ctx, key, existing); err != nil {
`

Reply:
UpdateStatus never got updated- should be fixed now. I can add a unit test for UpdateStatus in the upsert_test